pkg/instrumentation/podmutator.go

```diff
@@ -392,18 +394,37 @@ func (pm *instPodMutator) getInstrumentationInstance(ctx context.Context, ns cor
 	return otelInst, nil
 }
 
-func (pm *instPodMutator) selectInstrumentationInstanceFromNamespace(ctx context.Context, ns corev1.Namespace) (*v1alpha1.Instrumentation, error) {
+func (pm *instPodMutator) selectInstrumentationInstanceFromNamespace(ctx context.Context, ns corev1.Namespace, pod corev1.Pod) (*v1alpha1.Instrumentation, error) {
 	var otelInsts v1alpha1.InstrumentationList
 	if err := pm.Client.List(ctx, &otelInsts, client.InNamespace(ns.Name)); err != nil {
 		return nil, err
 	}
 
-	switch s := len(otelInsts.Items); {
+	// selector
+	var availableInstrument []v1alpha1.Instrumentation
+	for _, ins := range otelInsts.Items {
+		isMatch := true
+		if ins.Spec.Selector != nil {
+			labelSelector, err := v1.LabelSelectorAsSelector(ins.Spec.Selector)
+			if err != nil {
+				return nil, err
+			}
+			set := labels.Set(pod.GetLabels())
+			// selector not match
+			if !labelSelector.Matches(set) {
+				isMatch = false
+			}
+		}
+
+		if isMatch {
+			availableInstrument = append(availableInstrument, ins)
+		}
+	}
+
+	switch s := len(availableInstrument); {
 	case s == 0:
 		return nil, errNoInstancesAvailable
-	case s > 1:
-		return nil, errMultipleInstancesPossible
```

Comment:
This is a significant change in behavior. Until now, if we had multiple instrumentations applicable to the same Pod, we'd just error. After this change, we'd be applying one of them without clear criteria for the choice. I can see how this makes sense with the rest of your PR, as making sure your selector are mutually exclusive sounds annoying. 

For this PR, I'd like to keep this the way it was, and we can discuss changing it in a follow-up.

Reply:
Yes, this is a breaking change.

I'll create another thread to discuss this when this PR is complete.